overlay.nix

```diff
@@ -45,6 +45,12 @@ in
       cue = prev.callPackage ./pkgs/cue.nix {};
       devShell = final.callPackage ./pkgs/dev-shell.nix {};
       docker-distribution = prev.callPackage ./pkgs/docker-distribution.nix {};
+
+      # Pin docker and containerd to avoid unexpected cluster wide docker daemon restarts
+      # during metal deploy resulting in OCI jobs being killed or behaving unexpectedly
+      docker = (pkgsDocker final).docker; # v20.10.15
+      containerd = (pkgsDocker final).containerd; # v20.10.15
```

Comment:
Why is `final` passed here? We only want to evaluate `nixpkgs-docker` once, right?

Reply:
You're right, final vs. prev build hash is the same, eval/build time seems to be the same, but prev would be preferable as the right/better way to do it.